x-pack/plugins/triggers_actions_ui/public/application/components/builtin_action_types/swimlane/swimlane_connectors.tsx

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { Fragment, useCallback, useEffect, useMemo, useRef, useState } from 'react';
+import {
+  EuiCallOut,
+  EuiFieldText,
+  EuiFormRow,
+  EuiLink,
+  EuiSpacer,
+  EuiText,
+  EuiHorizontalRule,
+  EuiComboBox,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import * as i18n from './translations';
+import { ActionConnectorFieldsProps } from '../../../../types';
+import { SwimlaneActionConnector, SwimlaneConfig, SwimlaneFieldMap } from './types';
+import { useKibana } from '../../../../common/lib/kibana';
+import { getApplication } from './api';
+
+const SwimlaneActionConnectorFields: React.FunctionComponent<
+  ActionConnectorFieldsProps<SwimlaneActionConnector>
+> = ({ consumer, errors, action, editActionConfig, editActionSecrets, readOnly }) => {
+  const {
+    http,
+    notifications: { toasts },
+  } = useKibana().services;
+
+  const { docLinks } = useKibana().services;
+  const defaultNames: SwimlaneFieldMap[] = useMemo(
+    () => [
+      { key: 'alertSourceKeyName', name: 'alert-source' },
+      { key: 'severityKeyName', name: 'severity' },
+      { key: 'caseNameKeyName', name: 'case-name' },
+      { key: 'caseIdKeyName', name: 'case-id' },
+      { key: 'alertNameKeyName', name: 'alert-name' },
+      { key: 'commentsKeyName', name: 'comments' },
+    ],
+    []
+  );
+
+  const { apiToken } = useMemo(() => {
+    const secrets = action.secrets ?? {};
+
+    return secrets;
+  }, [action.secrets]);
+
+  const { mappings, apiUrl, appId } = useMemo(() => {
+    const config = action.config ?? {};
+
+    config.mappings =
+      config.mappings ??
+      (Object.assign(
+        {},
+        ...defaultNames.map((s) => ({ [s.key]: { key: s.name, id: '' } }))
```

Comment:
by looking at `SwimlaneFieldMappingConfig`, it seems that we want `fieldKey` and `fieldId`;